tests/test_single_topology_v3.py

```diff
@@ -273,3 +272,32 @@ def test_combine_masses():
     test_masses = st.combine_masses()
     ref_masses = [Br_mass, C_mass, C_mass, max(C_mass, N_mass), C_mass, C_mass, C_mass, F_mass]
     np.testing.assert_almost_equal(test_masses, ref_masses)
+
+
+def test_jit_intermediate_potential():
+    def setup_arbitary_transformation():
+        # NOTE: test system can probably be simplified; we just need
+        # any SingleTopologyV3 and conformation
+        with resources.path("timemachine.testsystems.data", "ligands_40.sdf") as path_to_ligand:
+            suppl = Chem.SDMolSupplier(str(path_to_ligand), removeHs=False)
+            mols = {get_mol_name(mol): mol for mol in suppl}
+
+        mol_a = mols["206"]
+        mol_b = mols["57"]
+
+        mcs_threshold = 0.75
+        res = atom_mapping.mcs_map(mol_a, mol_b, mcs_threshold)
+        query = Chem.MolFromSmarts(res.smartsString)
+        core_pairs = atom_mapping.get_core_by_mcs(mol_a, mol_b, query, mcs_threshold)
+
+        ff = Forcefield.load_from_file(DEFAULT_FF)
+        st = SingleTopologyV3(mol_a, mol_b, core_pairs, ff)
+        conf = st.combine_confs(get_romol_conf(mol_a), get_romol_conf(mol_b))
+        return st, conf
+
+    st, conf = setup_arbitary_transformation()
+
+    def U(lam):
+        return st.setup_intermediate_state(lam).get_U_fn()(conf)
+
+    jax.jit(U)(0.1)
```

Comment:
Nit: might expand this test slightly to exercise fast map over lambdas and or confs

```python
def U(x, lam):
   ...

confs = np.array([conf for _ in range(100)])
lams = np.linspace(0, 1, 100)

vmapped_U = jax.vmap(U, in_axes=(0,0))
_ = vmapped_U(confs, lams)
_ = jax.jit(vmapped_U)(confs, lams)
```

or similar

Reply:
Good call, added in 9018ca9.